### PR TITLE
adapter: Remove old migrations

### PR DESCRIPTION
**Pull request by jkosh44**
### Motivation
This PR refactors existing code.

### Checklist

- [X] This PR has adequate test coverage / QA involvement has been duly considered.
- [X] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] This PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way) and therefore is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  -

### CODE REVIEW

**Comment by jkosh44** [comment]
See: https://github.com/MaterializeInc/materialize/issues/17466

File: src/adapter/src/catalog/storage.rs [hunk 1]
```diff
@@ -315,75 +313,15 @@ async fn migrate(
...
-        },
-        // Attributes were added to role definitions.
-        //
-        // Introduced in v0.45.0
-        //
-        // TODO(jkosh44) Can be cleared (patched to be empty) in v0.46.0
-        |txn: &mut Transaction<'_>, _now, _bootstrap_args| {
-            txn.roles.update(|_role_key, role_value| {
-                let mut role_value = role_value.clone();
-                if role_value.role.attributes.is_none() {
```
Comment (mjibson) [inline src/adapter/src/catalog/storage.rs:375]: Can RoleAttribute be made into a non-Option now?

Comment (jkosh44) [reply src/adapter/src/catalog/storage.rs:375]: Done

**Comment by jkosh44** [comment]
It looks like the upgrade tests are testing versions earlier than v45, so we'll have to wait until the next release to merge.

File: src/environmentd/src/lib.rs [hunk 2]
```diff
@@ -348,12 +347,6 @@ pub async fn serve(config: Config) -> Result<Server, anyhow::Error> {
             .context("opening storage usage client")?,
     );
 
-    // TODO(teskje): Remove this migration in v0.42, since v0.41+ will only create orchestrator
-    // resources in the "cluster" namespace.
-    tracing::info!("SPECIAL MIGRATION: removing legacy orchestrator services");
-    remove_orchestrator_services(config.controller.orchestrator.namespace("compute")).await?;
-    remove_orchestrator_services(config.controller.orchestrator.namespace("storage")).await?;
```
Comment (teskje) [inline src/environmentd/src/lib.rs:355]: I forgot about this, thanks!

**Review by teskje — APPROVED** [review]
LGTM!

**Comment by jkosh44** [comment]
It looks like the upgrade tests are testing versions earlier than v45, so we'll have to wait until the next release to merge.

> It looks like the upgrade tests are testing versions earlier than v45, so we'll have to wait until the next release to merge.

Turns out we have to wait for one more release: https://materializeinc.slack.com/archives/C01LKF361MZ/p1678816536147819

**Pull request merged**